llvm/lib/Target/AMDGPU/SIPreAllocateWWMRegs.cpp

```diff
@@ -254,3 +262,13 @@ bool SIPreAllocateWWMRegs::runOnMachineFunction(MachineFunction &MF) {
   rewriteRegs(MF);
   return true;
 }
+
+PreservedAnalyses
+SIPreAllocateWWMRegsPass::run(MachineFunction &MF,
+                              MachineFunctionAnalysisManager &MFAM) {
+  auto *LIS = &MFAM.getResult<LiveIntervalsAnalysis>(MF);
+  auto *Matrix = &MFAM.getResult<LiveRegMatrixAnalysis>(MF);
+  auto *VRM = &MFAM.getResult<VirtRegMapAnalysis>(MF);
```

Comment:
I thought this should be using getCachedResult, and the pass supported no LIS for the fast RA path. But I see now the legacy path is requiring them (although it probably shouldn't?)

Reply:
Yep, this looks odd. We shouldn't involve VRM for the fast regalloc case.